### PR TITLE
Add new darc to omniledger

### PR DESCRIPTION
**Pull request by kc1212**
The current version is not complete, but in a usable state. Concretely,
it does not support online evolution. There are also some minor tests
missing. Nevertheless, the API will not change significantly so we can
start to integrate it into omniledger.

### CODE REVIEW

File: omniledger/darc/darc.go [hunk 1]
```diff
@@ -1,103 +1,82 @@
 /*
-Package darc in most of our projects we need some kind of access control to
-protect resources. Instead of having a simple password
+Package darc in most of our projects we need some kind of access control to protect resources. Instead of having a simple password
```
Comment (ineiti) [inline omniledger/darc/darc.go]: nitpick: line-limit of 80 caracters

File: omniledger/darc/darc.go [hunk 2]
```diff
@@ -118,662 +97,745 @@ func NewDarcFromProto(protoDarc []byte) *Darc {
...
+	return nil
+}
+
+// EvolveOnline works like Evolve, but doesn't inlcude all the necessary data
+// to verify the update in an offline setting. This is enough for the use case
+// where the ocs stores all darcs in its internal database.  The service
+// verifying the signature will have to verify if there is a valid path from
+// the previous darc to the signer.
+func (d *Darc) EvolveOnline(prevd *Darc, prevOwner *Signer) error {
+	/*
```
Comment (ineiti) [inline omniledger/darc/darc.go]: Is this dead code and can be removed?

Comment (kc1212) [reply omniledger/darc/darc.go]: it's to remind me that EvolveOnline still needs to be done

**Comment by ineiti** [comment]
Wow - more than 1000 new lines... Is it OK if I look at it while using it? One thing: can you add a `doc.go` with a description of some of the basic concepts and some examples how to use it?

**Pull request merged**